src/handlers/transfers/handler.js

```diff
@@ -212,7 +228,7 @@ const prepare = async (error, messages) => {
         Logger.isErrorEnabled && Logger.error(Util.breadcrumb(location, { path: 'validationFailed' }))
         try {
           Logger.isInfoEnabled && Logger.info(Util.breadcrumb(location, 'saveInvalidRequest'))
-          await TransferService.prepare(payload, reasons.toString(), false)
+          await TransferService.prepare(message, reasons.toString(), false)
```

Comment:
I think I missed this in my commit Vijay. 
Hmm...I guess we might have to fetch the settlement model regardless of validation since we save the transfer even if it failed validation.